### PR TITLE
Replace getallcolumn function callback by a structure similar to boun…

### PR DESCRIPTION
**Pull request by steadtler**
…dexpression thats going to be a bit more flexible

### CODE REVIEW

File: sql/sql_expression.h [hunk 1]
```diff
@@ -495,6 +495,39 @@ struct GetAllColumnsOutput {
...
+/** Filter that rejects, accepts and or/modify a columns name.
+*/
+
+struct ColumnFilter {
+    typedef std::function<ColumnName (const ColumnName &)> Exec;
+   
+    ColumnFilter() 
+    {
+
+    }
```
Comment (FinchPowers) [inline sql/sql_expression.h]: Ça sert à quoi de le déclarer s'il est vide?

**Comment by FinchPowers** [comment]
+1

File: sql/sql_expression.h (hunk 1 above)
Comment (steadtler) [inline sql/sql_expression.h]: Pour ne pas que le compilateur se plaigne que tu n'utilise pas l'autre.

Comment (FinchPowers) [inline sql/sql_expression.h]: Ah ben, je ne savais pas qu'il s'en plaindrait... :)

File: sql/sql_expression.h [hunk 2]
```diff
@@ -495,6 +495,39 @@ struct GetAllColumnsOutput {
...
+    operator bool () const { return !!exec; }
+
+    Exec exec;
+
+    ColumnName operator () (const ColumnName & columnName) const
+    {
+        if (exec)
+            return exec(columnName);
+        else
+            return columnName;
```
Comment (guyd) [inline sql/sql_expression.h]: Consider creating a static member of type ColumnFilter named _identity_ to do just that.  This way the code is very explicit and we can prevent errors using a ColumnFilter that has wrongly not been initialized.

**Comment by jeremybarnes** [comment]
Should be a const reference to the column filter.

**Pull request merged**